lib/chef/http/basic_client.rb

```diff
@@ -103,14 +103,17 @@ def proxy_uri
         # Check if the proxy string contains a scheme. If not, add the url's scheme to the
         # proxy before parsing. The regex /^.*:\/\// matches, for example, http://. Reusing proxy
         # here since we are really just trying to get the string built correctly.
-        if String === proxy && !proxy.strip.empty?
+        if proxy.is_a?(String) && !proxy.strip.empty?
           if proxy.match(/^.*:\/\//)
-           proxy = URI.parse(proxy.strip)
+            proxy = URI.parse(proxy.strip)
           else
-           proxy = URI.parse("#{url.scheme}://#{proxy.strip}")
-          end 
+            proxy = URI.parse("#{url.scheme}://#{proxy.strip}")
+          end
+        # If proxy is anything other than a non-empty string, set it to nil
+        else
+          proxy = nil
```

Comment:
It's initially set on line 100, where it could end up being an empty string, which is what happened to the reporter.